src/client-side-encryption/state_machine.ts

```diff
@@ -519,16 +521,16 @@ export class StateMachine {
   ): Promise<Uint8Array | null> {
     const { db } = MongoDBCollectionNamespace.fromString(ns);
 
-    const collections = await client
-      .db(db)
-      .listCollections(filter, {
-        promoteLongs: false,
-        promoteValues: false,
-        ...(timeoutContext?.csotEnabled()
-          ? { timeoutMS: timeoutContext?.remainingTimeMS, timeoutMode: 'cursorLifetime' }
-          : {})
-      })
-      .toArray();
+    const cursor = new ListCollectionsCursor(client.db(db), filter, {
```

Comment:
Doesn't calling` client.db(db).listCollections(...).toArray` have the same behavior?

Reply:
If we call `listCollections()`, we'd need to add logic to support `timeoutContext` as a public API option.  I can do that if you'd prefer!

Reply:
If we keep `listCollections` as is, will it support `timeoutMS` reset upon cursor refresh?

Reply:
I was looking at `resolveOptions` and thought that we needed to add `timeoutContext` support so that the option would be inherited, but I was wrong.  I reverted this change